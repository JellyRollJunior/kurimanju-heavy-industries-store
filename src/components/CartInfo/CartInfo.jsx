import styles from './CartInfo.module.css';

const CartInfo = ({ numCartItems = 0, subtotal = 0, checkoutCart }) => {
  const formattedSubtotal = subtotal.toLocaleString('en-US').replace(/,/g, ' ');

  return (
    <section>
      <h3 className={styles.subtotal}>Subtotal ({numCartItems} items):</h3>
      <h3 className={styles.subtotal}>$ {formattedSubtotal}</h3>
      <button onClick={checkoutCart} className={styles.checkoutButton}>Checkout</button>
    </section>
  );
};

export { CartInfo };
