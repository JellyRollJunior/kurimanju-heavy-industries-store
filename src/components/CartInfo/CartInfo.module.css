.subtotal {
  composes: tertiaryColor from '../../styles/colors.module.css';
}

.checkoutButton {
  width: 100%;
}
