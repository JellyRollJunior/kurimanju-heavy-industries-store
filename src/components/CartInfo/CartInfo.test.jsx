import { render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import userEvent from '@testing-library/user-event';
import { CartInfo } from './CartInfo';

describe('Cart info component', () => {
  it('Renders cart item quantity, subtotal, and checkout button', () => {
    const { container } = render(<CartInfo numCartItems={5} subtotal={10} />);

    expect(container).toMatchSnapshot();
  });

  it('Calls checkoutCart when checkout button is clicked', async () => {
    const checkoutCart = vi.fn();
    render(<CartInfo subtotal={10} checkoutCart={checkoutCart} />);
    const user = userEvent.setup();

    await user.click(screen.getByRole('button', 'Checkout'));

    expect(checkoutCart).toBeCalled();
  });
});
