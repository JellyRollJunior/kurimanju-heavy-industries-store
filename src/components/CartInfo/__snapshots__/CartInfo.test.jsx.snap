// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cart info component > Renders cart item quantity, subtotal, and checkout button 1`] = `
<div>
  <section>
    <h3
      class="_subtotal_2d26a7"
    >
      Subtotal (
      5
       items):
    </h3>
    <h3
      class="_subtotal_2d26a7"
    >
      $ 
      10
    </h3>
    <button
      class="_checkoutButton_2d26a7"
    >
      Checkout
    </button>
  </section>
</div>
`;
